File: src/main/java/com/fendyk/clients/redis/RedisMinecraftUser.java
package com.fendyk.clients.redis;

import com.fendyk.DTOs.MinecraftUserDTO;
import com.fendyk.Main;
import com.fendyk.clients.RedisAPI;
import io.lettuce.core.RedisClient;
import io.lettuce.core.pubsub.RedisPubSubListener;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.UUID;

public class RedisMinecraftUser extends RedisAPI<MinecraftUserDTO> {
    public RedisMinecraftUser() {
        super("minecraftuser:", MinecraftUserDTO.class);
    }

}

File: src/main/java/com/fendyk/clients/redis/RedisMinecraftUser.kt
package com.fendyk.clients.redis

import com.fendyk.DTOs.MinecraftUserDTO
import com.fendyk.clients.RedisAPI
import java.util.UUID
import java.util.concurrent.CompletableFuture

class RedisMinecraftUser : RedisAPI<UUID, MinecraftUserDTO>("minecraftuser:", MinecraftUserDTO::class.java) {
    override fun get(key: UUID): CompletableFuture<MinecraftUserDTO> {
        return fetch(key.toString())
    }
    override fun set(key: UUID, dto: MinecraftUserDTO): CompletableFuture<Boolean> {
        return save(key.toString(), dto)
    }
}
